src/components/NewClusterModal.js

```diff
@@ -185,22 +185,181 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
       options: _.map(({ label, image }) => ({ label, value: image }), leoImages)
     })
 
+    const makeGroupedEnvSelect = id => h(GroupedSelect, {
+      id,
+      value: selectedLeoImage,
+      onChange: ({ value }) => {
+        this.setState({ selectedLeoImage: value })
+      },
+      isSearchable: false,
+      isClearable: false,
+      options: [{ label: 'JUPYTER ENVIRONMENTS', options: _.map(({ label, image }) => ({ label, value: image }), leoImages) },
+        { label: 'OTHER ENVIRONMENTS', options: [{ label: 'Custom Environment', value: CUSTOM_MODE }] }]
+    })
+
     const makeImageInfo = style => div({ style: { whiteSpace: 'pre', ...style } }, [
       div({ style: Style.proportionalNumbers }, ['Updated: ', updated ? Utils.makeStandardDate(updated) : null]),
       div(['Version: ', version || null])
     ])
 
-    const { contents, onPrevious } = Utils.switchCase(viewMode, [
-      'Packages', () => ({
-        onPrevious: () => this.setState({ viewingPackages: false, viewMode: undefined }),
+    const bottomButtons = () => h(Fragment, [
+      div({ style: { flexGrow: 1 } }),
+      div(
+        { style: { display: 'flex', marginTop: '1rem' } },
+        [
+          div({ style: { marginLeft: 'auto', marginRight: '2rem' } }, [
+            h(ButtonSecondary, {
+              onClick: onDismiss
+            }, 'Cancel')
+          ]),
+          div([
+            h(ButtonPrimary, {
+              disabled: selectedLeoImage === CUSTOM_MODE && isCustomImageInvalid,
+              tooltip: selectedLeoImage === CUSTOM_MODE && isCustomImageInvalid &&
```

Comment:
It might be nice to extract `selectedLeoImage === CUSTOM_MODE` to a local variable, since it's repeated several times throughout.